test/e2e/annotations/globalratelimit.go

```diff
@@ -72,12 +73,14 @@ var _ = framework.DescribeAnnotation("annotation-global-rate-limit", func() {
 			return true
 		})
 		assert.Contains(ginkgo.GinkgoT(), serverConfig,
-			fmt.Sprintf(`global_throttle = { namespace = "%v", `+
+			fmt.Sprintf(`global_throttle = { namespace = "%v", set_headers = true, `+
 				`limit = 5, window_size = 120, `+
 				`key = { { nil, "remote_addr", nil, nil, }, { nil, "http_x_api_client", nil, nil, }, }, `+
 				`ignored_cidrs = { "192.168.1.1", "234.234.234.0/24", } }`,
 				namespace))
 
 		f.HTTPTestClient().GET("/").WithHeader("Host", host).Expect().Status(http.StatusOK)
+
```

Comment:
Let's also add some assertions on the returned ratelimit headers. You can refer to this for how to do that: https://github.com/kubernetes/ingress-nginx/blob/2555747a2739253189c3af5a3db28ce216af25c1/test/e2e/annotations/auth.go#L430

Reply:
Added assertions as suggested